From d58a1a1fd812df24c0ff8d844daabba754cded2b Mon Sep 17 00:00:00 2001
From: Stephano Cetola <stephano@cetola.net>
Date: Thu, 9 Jul 2026 18:14:31 +0000
Subject: [PATCH 11/20] arm: rk3588s: pocket-reform-rcm5: enable DSI1 panel

Enable the MIPI DSI1 display pipeline in barebox for the Radxa CM5
(RK3588S2) board.

Add hdptxphy0 + display_subsystem clock, vop, vop_mmu, mipidcphy1,
dsi1 (with JDI panel), the VOP2 viewport endpoint (vp2 -> MIPI1 ->
dsi1), and the backlight-enable pinctrl group. VOP2 lists hdptxphy0
as clock pll_hdmiphy0 (index 7). Without the PHY enabled barebox
permanently defers VOP2 probe. RK3588S has only hdptxphy0 (no
hdptxphy1 as on the full RK3588).

Signed-off-by: Stephano Cetola <stephano@cetola.net>
---
 .../dts/rk3588s-mnt-pocket-reform-rcm5.dts   | 70 +++++++++++++++++++
 1 file changed, 70 insertions(+)

diff --git a/arch/arm/dts/rk3588s-mnt-pocket-reform-rcm5.dts b/arch/arm/dts/rk3588s-mnt-pocket-reform-rcm5.dts
index a3c5d20bd9..8c4ea52d63 100644
--- a/arch/arm/dts/rk3588s-mnt-pocket-reform-rcm5.dts
+++ b/arch/arm/dts/rk3588s-mnt-pocket-reform-rcm5.dts
@@ -8,6 +8,7 @@

 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/pinctrl/rockchip.h>
+#include <dt-bindings/soc/rockchip,vop2.h>
 #include "arm64/rockchip/rk3588s.dtsi"
 #include "arm64/rockchip/rk3588s-radxa-cm5.dtsi"
 #include "rk3588.dtsi"
@@ -211,6 +212,75 @@
 				<1 RK_PD4 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	disp {
+		disp_en: disp-en {
+			rockchip,pins = <1 RK_PC2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+};
+
+&hdptxphy0 {
+	status = "okay";
+};
+
+&display_subsystem {
+	clocks = <&hdptxphy0>;
+	clock-names = "pll_hdmiphy0";
+};
+
+&vop_mmu {
+	status = "okay";
+};
+
+&vop {
+	status = "okay";
+};
+
+&mipidcphy1 {
+	status = "okay";
+};
+
+&dsi1 {
+	status = "okay";
+	rockchip,lane-rate = <972000>;
+
+	panel@0 {
+		compatible = "jdi,lt070me05000";
+		reg = <0>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&disp_en>;
+
+		/* backlight enable; reset and DCDC are driven by the RP2040 */
+		enable-gpios = <&gpio1 RK_PC2 GPIO_ACTIVE_HIGH>;
+		probe-mode-lpm;
+
+		port {
+			mipi_panel_in: endpoint {
+				remote-endpoint = <&dsi1_out_panel>;
+			};
+		};
+	};
+};
+
+&dsi1_in {
+	dsi1_in_vp2: endpoint {
+		remote-endpoint = <&vp2_out_dsi1>;
+	};
+};
+
+&dsi1_out {
+	dsi1_out_panel: endpoint {
+		remote-endpoint = <&mipi_panel_in>;
+	};
+};
+
+&vp2 {
+	vp2_out_dsi1: endpoint@ROCKCHIP_VOP2_EP_MIPI1 {
+		reg = <ROCKCHIP_VOP2_EP_MIPI1>;
+		remote-endpoint = <&dsi1_in_vp2>;
+	};
 };

 /* Override the barebox rk3588.dtsi SCMI rates to also set SPLL to 702 MHz,
--
2.43.0
